### PR TITLE
all: rename to use candid instead of idm

### PR DESCRIPTION
**Pull request by rogpeppe**
For the time being, we leave the special @idm domain
unchanged, as it will require special treatment to work
with existing databases and external agents.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -99,21 +99,21 @@ version/init.go: version/init.go.tmpl FORCE
...
-	GOBIN=$(CURDIR)/identity-$(GIT_VERSION)/bin go install $(INSTALL_FLAGS) -v $(PROJECT)/...
-	mv identity-$(GIT_VERSION)/bin/idserver identity-$(GIT_VERSION)/bin/identity
-	cp -r $(CURDIR)/templates identity-$(GIT_VERSION)
-	cp -r $(CURDIR)/static identity-$(GIT_VERSION)
-	tar cv identity-$(GIT_VERSION) | xz > $@
-	-rm -r identity-$(GIT_VERSION)
+candid-$(GIT_VERSION).tar.xz: version/init.go
+	mkdir -p candid-$(GIT_VERSION)/bin
+	GOBIN=$(CURDIR)/candid-$(GIT_VERSION)/bin go install $(INSTALL_FLAGS) -v $(PROJECT)/...
+	mv candid-$(GIT_VERSION)/bin/candidsrv candid-$(GIT_VERSION)/bin/candidsrv
```
Comment (mhilton) [inline Makefile]: Well this line doesn't do anything

File: Makefile [hunk 2]
```diff
@@ -99,21 +99,21 @@ version/init.go: version/init.go.tmpl FORCE
...
 # Build a release tarball
-identity-$(GIT_VERSION).tar.xz: version/init.go
-	mkdir -p identity-$(GIT_VERSION)/bin
-	GOBIN=$(CURDIR)/identity-$(GIT_VERSION)/bin go install $(INSTALL_FLAGS) -v $(PROJECT)/...
-	mv identity-$(GIT_VERSION)/bin/idserver identity-$(GIT_VERSION)/bin/identity
-	cp -r $(CURDIR)/templates identity-$(GIT_VERSION)
-	cp -r $(CURDIR)/static identity-$(GIT_VERSION)
-	tar cv identity-$(GIT_VERSION) | xz > $@
-	-rm -r identity-$(GIT_VERSION)
+candid-$(GIT_VERSION).tar.xz: version/init.go
```
Comment (mhilton) [inline Makefile]: This whole thing is really only useful for the charm build so it might be better stored with the charm code.

File: README.md [hunk 3]
```diff
@@ -40,13 +40,13 @@ At this point, from the root of this branch, run the command::
...
+- candidsrv: start the Candid identity server;
 
-## Identity server
+## Candid server
 
-The identity server can be started with the following command:
+The server can be started with the following command:
 
-    idserver -logging-config INFO cmd/idserver/config.yaml
+    candidsrv -logging-config INFO cmd/candidsrv/config.yaml
```
Comment (mhilton) [inline README.md]: Please update the config.yaml so that this will work for someone trying it.

File: docs/configuration.md [hunk 4]
```diff
@@ -10,7 +10,7 @@ Configuration File
 ------------------
 The identity manager loads its configuration at startup from a yaml
 file. In a usual installation this file is stored in
-/etc/blues-identity/config.yaml. An example configuration file is:
```
Comment (mhilton) [inline docs/configuration.md]: This document is probably massively out of date. Perhaps we should try and address that.

File: snap/idm/snapcraft.yaml [hunk 5]
```diff
@@ -1,33 +0,0 @@
-name: idm
```
Comment (mhilton) [inline snap/idm/snapcraft.yaml]: These appear to be deleted, you might as well remove the snap directory at the same time.

File: Makefile (hunk 2 above)
Comment (rogpeppe) [reply Makefile]: Let's do this one stage at a time. Perhaps better to remove it later?

File: Makefile (hunk 1 above)
Comment (rogpeppe) [reply Makefile]: good point

File: Makefile [hunk 6]
```diff
@@ -42,14 +42,14 @@ build: version/init.go
...
-release: identity-$(GIT_VERSION).tar.xz
+release: candid-$(GIT_VERSION).tar.xz
 
 install: version/init.go
 	go install $(INSTALL_FLAGS) -v $(PROJECT)/...
 
 clean:
-	$(MAKE) -C snap/idm clean
-	$(MAKE) -C snap/user-admin clean
+	$(MAKE) -C snap/candidsrv clean
```
[the remark this answers is not in the document]
Comment (rogpeppe) [reply Makefile]: I think I'll leave that for another PR.

File: README.md (hunk 3 above)
Comment (rogpeppe) [reply README.md]: The config file is still as valid as it was (it does still work as is), so I'll leave it for now.

File: docs/configuration.md (hunk 4 above)
Comment (rogpeppe) [reply docs/configuration.md]: TODO...

**Comment by rogpeppe** [comment]
:shipit:

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/candid/1/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/candid/11/

**Comment by fabricematrat** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.theblues.io:8080/job/candid-merge

**Comment by fabricematrat** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.theblues.io:8080/job/candid-merge

**Pull request merged**